newIDE/app/src/Utils/memoize.js

```diff
@@ -0,0 +1,21 @@
+// @flow
+
+export default function<Input, Output>(func: (Input) => Output): (Input) => Output {
+  const primitives = new Map();
+  const objects = new WeakMap();
+
+  function cacheFor<Input, Output>(input: Input): Output {
```

Comment:
This typing is not correct, you're not returning an output:
![image](https://user-images.githubusercontent.com/1280130/94280118-f0246f00-ff44-11ea-8851-96a39d85c715.png)
I think you can just remove `: Output` here and let the type checker infer the type.

Run `npm run flow` to run the static typing checker: https://github.com/4ian/GDevelop/tree/master/newIDE#tests